codalab/server/bundle_manager.py

```diff
@@ -355,26 +350,22 @@ def _filter_and_sort_workers(self, workers_list, bundle):
             worker_id = worker['worker_id']
             has_gpu[worker_id] = worker['gpus'] > 0
 
-        # deduct worker resources based on running bundles
-        self._deduct_worker_resources(workers_list)
-
         # Filter by CPUs.
-        request_cpus = self._compute_request_cpus(bundle)
-        if request_cpus:
-            workers_list = [worker for worker in workers_list if worker['cpus'] >= request_cpus]
+        workers_list = [
+            worker for worker in workers_list if worker['cpus'] >= bundle_resources.cpus
+        ]
 
         # Filter by GPUs.
-        request_gpus = self._compute_request_gpus(bundle)
-        if request_gpus:
-            workers_list = [worker for worker in workers_list if worker['gpus'] >= request_gpus]
-
-        # Filter by memory.
-        request_memory = self._compute_request_memory(bundle)
-        if request_memory:
+        if bundle_resources.gpus:
             workers_list = [
-                worker for worker in workers_list if worker['memory_bytes'] >= request_memory
+                worker for worker in workers_list if worker['gpus'] >= bundle_resources.gpus
             ]
 
+        # Filter by memory.
```

Comment:
There's no `free-for-all` workers anymore so all runs request some amount of memory and cpu by default.

Reply:
Also I just realized that before this change, we actually were not filtering for memory when the request memory was the default value (ie 2gb). Shouldn't have been a practical issues since our workers rarely run out of memory but still, might've caused issues in the future.